datadog.go

```diff
@@ -0,0 +1,214 @@
+// Package datadog contains a Datadog exporter.
+//
+// This exporter is currently work in progress
+package datadog 
+// import "go.opencensus.io/exporter/datadog"
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"log"
+	"sync"
+
+	//"go.opencensus.io/internal"
+	"go.opencensus.io/stats/view"
+	"go.opencensus.io/tag"
+	"github.com/DataDog/datadog-go/statsd"
+)
+
+// Exporter exports stats to Datadog
+type Exporter struct{
+	opts		Options
+	c			*collector
+	d			*statsd.Client
+}
+
+// Options contains options for configuring the exporter
+type Options struct {
+	// Namespace to prepend to all metrics
+	Namespace 	string
+
+	// Endpoint
+	Endpoint 	string
+
+	// OnError is the hook to be called when there is
+	// an error occurred when uploading the stats data.
+	// If no custom hook is set, errors are logged.
+	// Optional.
+	OnError		func(err error)
+
+	// Tags are global tags added to each metric
+	Tags	[]string
+}
+
+var (
+	newExporterOnce sync.Once
+	errSingletonExporter = errors.New("expecting only one exporter per instance")
+)
+
+// NewExporter returns an exporter that exports stats to Datadog
+func NewExporter(o Options) (*Exporter, error) {
+	var err = errSingletonExporter
+	var exporter *Exporter
+	newExporterOnce.Do(func() {
+		exporter, err = newExporter(o)
+	})
+	return exporter, err
+}
+
+func newExporter(o Options) (*Exporter, error) {
+	endpoint := o.Endpoint
+	if endpoint == "" {
+		endpoint = "127.0.0.1:8125"
```

Comment:
let's make the StatsD host and port separate options so we have less errors to handle